### PR TITLE
[ELY-1061] Specify parameter requirements on HttpServerAuthenticationMechanismFactory.

### PR DESCRIPTION
**Pull request by darranl**

### CODE REVIEW

File: src/main/java/org/wildfly/security/http/HttpServerAuthenticationMechanismFactory.java [hunk 1]
```diff
@@ -32,20 +32,20 @@
      * Get the names of the HTTP authentication mechanisms that can be supplied by this factory filtered by the supplied
      * properties.
      *
-     * @param properties the selection criteria for the mechanisms.
-     * @return the names of the supported HTTP authentication mechanisms.
+     * @param properties the {@code non-null} set of properties to pass configuration to the mechanisms that may be evaluated for mechanism availability.
```
Comment (mchoma) [inline src/main/java/org/wildfly/security/http/HttpServerAuthenticationMechanismFactory.java]: Isn't it better to allow null here?:
- SaslClientFactory.getMechanismNames allow null
- SaslServerFactory.getMechanismNames allow null
- ServerMechanismFactoryImpl.getMechanismNames accept null

Comment (darranl) [inline src/main/java/org/wildfly/security/http/HttpServerAuthenticationMechanismFactory.java]: Am happy with non-null here - we always expect it and it saves implementations having to handle null differently.

Comment (mchoma) [inline src/main/java/org/wildfly/security/http/HttpServerAuthenticationMechanismFactory.java]: So in your PoV,  the fact that `ServerMechanismFactoryImpl.getMechanismNames accepts null` is not breaking the contract? Shouldn't ServerMechanismFactoryImpl.getMechanismNames validates properties if they are really non-null.

Comment (darranl) [inline src/main/java/org/wildfly/security/http/HttpServerAuthenticationMechanismFactory.java]: Specifying that callers to an API must not pass in null does not mandate an implementation throws an error.

**Comment by msve** [comment]
Downstream jira: https://issues.jboss.org/browse/JBEAP-7553

**Pull request merged**